### DISCUSSION TITLE
变量初值设置

### DISCUSSION THREAD

**Discussion by huiq666** [metab0t/PyOptInterface#12] [中文Q&A] [answered]
尊敬的PyOptInterface团队您好！
我想请教有关变量处置设置的问题。在pyomo中我通过以下代码固定其指定的变量不变（这里我固定其值为0和1），我想知道如何使用PyOptInterface来实现相应的功能。
`for j in range(1,J+1):`
    `if table_A[j-1,4] > 0:`
        `model.var_v[j,` `0].fix(1)`
    `elif` table_A[j-1,4] <= `0:`
        `model.var_v[j,` `0].fix(0)`

**Comment by metab0t** [comment, marked as answer]
Set the upper bound and lower bound of the variable to the same value.
```python
def fix_variable(model, var, value):
    model.set_variable_attribute(var, poi.VariableAttribute.LowerBound, value)
    model.set_variable_attribute(var, poi.VariableAttribute.UpperBound, value)

fix_variable(model, x, 1.0)
```

**Reply by huiq666** [reply]
Thank you very much for your answer.

**Comment by huiq666** [comment]
尊敬的pyoptinterface团队您好！
我想知道如何获取变量求解前的初始值相关问题。我在pyomo上设置了一个变量对其进行赋初值后又将这个变量拿出来使用以下是我的相关代码：
`model.var_v = Var(range(1,J+1), range(0,K+1), within=Binary)`
`for j in range(1,J+1):
    if table_A[j-1,4] > 0:
        model.var_v[j, 0].fix(1)
    elif table_A[j-1,4] <= 0:
        model.var_v[j, 0].fix(0)`
`initialvalue_G = np.copy(table_A[:, 2] - table_A[:, 4])
for j in range(1, J+1):
    initialvalue_G[j-1] = initialvalue_G[j-1] * value(model.var_v[j, 0])`
  我想请问如何使用pyoptinterface实现value(model.var_v[j, 0])的相关功能。
  谢谢您！
  
  2、我想知道如何查询我定义了那些变量和约束条件的形式，用于检查代码是否有误，该如何写相关代码？

**Comment by metab0t** [comment]
Please use codeblock in markdown to make your code more readable for your future comments.

1. Use `model.get_variable_attribute(x, poi.VariableAttribute.LowerBound` to query information of variable.
2. Use [`pprint`](https://metab0t.github.io/PyOptInterface/expression.html#pretty-print-expression) to show expressions, or use [`write`](https://metab0t.github.io/PyOptInterface/model.html#write-the-model-to-file) to export your model to text file. Remember to set the name of variable and constraint correctly.
